### PR TITLE
[URGENT] Misc. server fixes

### PR DESCRIPTION
**Pull request by bls1999**
### Things that work again
 - Setting the Campaign
 - Level deletion (saving/loading works as intended in 152.3)
 - Chat auto anti-spam filter
 - Banning (and bans applying to users)
 - Unpublishing levels
 - Trial/perma mod detection
 - PMs tab pages
 - Ban edit mod logging (doesn't log the mod's IP address as yes/no)
 - All server restarts
 - Blank level listing pages (don't throw an error)
 - Guild invitations/leaving
 - Artifact placement
 - Old account deletion (monthly cron task; find in weekly.php)
 - Vault

### Five things require your attention in addition to merging this
 - All of the level music is gone from PR2hub (404 error)
 - Tournament server is down
 - None of the Campaign levels have prizes (restore your last backup of the pr2_campaigns table and just rename it to campaigns)
 - Please restore the recent_logins table. This will continue to help us do our jobs diligently [with IP comparison], and will also allow us to verify more account recoveries. With PRF on the horizon, we want to be able to process as many of these as we can. :pray:
 - Please make sure the vault works!
 - Please run this query on the database (future PR2 sets smile upon you):
```mysql
ALTER TABLE pr2
  CHANGE hat_array hat_array varchar(1000),
  CHANGE head_array head_array varchar(1000),
  CHANGE body_array body_array varchar(1000),
  CHANGE feet_array feet_array varchar(1000);

ALTER TABLE epic_upgrades
  CHANGE epic_hats epic_hats varchar(1000),
  CHANGE epic_heads epic_heads varchar(1000),
  CHANGE epic_bodies epic_bodies varchar(1000),
  CHANGE epic_feet epic_feet varchar(1000);
```

### Future
 - Switch the client back to GET for messages_get.php and guild_leave.php so default_get can be used. Require explicit token for both
___

See the client repo for v152.3

### CODE REVIEW

**Comment by EternalUF** [comment]
URGENT @jacob-grahn

**Comment by EternalUF** [comment]
Urgentest

File: http_server/logout.php [hunk 1]
```diff
@@ -24,7 +24,9 @@
     // client has token, token sent via post
     if (!is_empty($_POST['token']) && is_empty($data)) {
         $ret->errorType = 'token';
-        token_delete($pdo, $token);
+        if (token_delete($pdo, $token, true) === false) {
```
Comment (TheMightyBacon1) [inline http_server/logout.php]: The exploit being fixed: https://gitlab.com/TheMightyBacon/pr2logoutexploitbypass1

**Comment by EternalUF** [comment]
@jacob-grahn

**Pull request merged**